src/services/walletconnect/WalletConnectContext.tsx

```diff
@@ -42,6 +45,13 @@ export const WalletConnectProvider = ({ children }: { children: ReactNode }) =>
     walletConnect.updateSessions(chainId, safeAddress).catch(setError)
   }, [walletConnect, chainId, safeAddress])
 
+  // Disconnect all sessions if no Safe is loaded
+  useEffect(() => {
+    if (!walletConnect || isSafeConnected) return
+
+    walletConnect.disconnectAllSessions().catch(setError)
```

Comment:
What if a Safe is just being loaded? WC sessions live across page reloads unless disconnected.

Reply:
We [reference the `safe` search param](https://github.com/safe-global/safe-wallet-web/pull/2640/files#diff-d8b8f8e4abf39b33900def2ffb040c4e238c797d2b4281f97a1d7992d7f3fa6bR31) for this reason.